models/demos/mnist/demo/demo.py

```diff
@@ -0,0 +1,78 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+import torch
+import ttnn
+
+from torchvision import transforms, datasets
+from loguru import logger
+
+from torch.utils.data import DataLoader
+from models.demos.mnist.reference.mnist import MnistModel
+from models.demos.mnist.tt import tt_mnist
+
+from ttnn.model_preprocessing import preprocess_model_parameters
+
+
+def run_demo_dataset(device, batch_size, iterations, model_location_generator):
+    transform = transforms.Compose([transforms.ToTensor()])
+    test_dataset = datasets.MNIST(root="./data", train=False, transform=transform, download=True)
+
+    state_dict = torch.load(model_location_generator("mnist_model.pt", model_subdir="mnist"))
+    model = MnistModel(state_dict)
+    model = model.eval()
+
+    parameters = preprocess_model_parameters(
+        initialize_model=lambda: model,
+        convert_to_ttnn=lambda *_: True,
+        device=device,
+    )
+    correct = 0
+    for iters in range(iterations):
+        dataloader = DataLoader(test_dataset, batch_size=batch_size)
+        x, labels = next(iter(dataloader))
+        dataset_predictions = []
+        ttnn_predictions = []
+        dataset_ttnn_correct = 0
+        x = ttnn.from_torch(x, dtype=ttnn.bfloat16, device=device)
+        tt_output = tt_mnist.mnist(device, batch_size, x, parameters)
+        tt_output = ttnn.to_torch(tt_output)
+        predicted_probabilities = torch.nn.functional.softmax(tt_output, dim=1)
+        _, predicted_label = torch.max(predicted_probabilities, 1)
+        tt_output = tt_output
```

Comment:
Is this line needed?

Reply:
Thank you for pointing that out! I’ve removed the redundant line .